src/rlookup.c

```diff
@@ -599,10 +599,22 @@ static RSValue *replyElemToValue(RedisModuleCallReply *rep, RLookupCoerceType ot
   }
 }
 
+static bool isValueAvailable(const RLookupKey *kk, const RLookupRow *dst, RLookupLoadOptions *options) {
+  if (!options->forceLoad) {
+    if (kk->flags & RLOOKUP_F_VAL_AVAILABLE) {
+      // No need to "write" this key. It's always implicitly loaded!
+      return true;
+    } else if ((kk->flags & RLOOKUP_F_SVSRC) && (RLookup_GetItem(kk, dst) == NULL)) {
+      // There is no value in the sorting vector, and we don't need to load it from the document.
+      return true;
+    }
+  }
+  return false;
```

Comment:
Consider avoiding `if (cond) return true; else return false;` pattern
```suggestion
  return (!options->forceLoad && (
      // No need to "write" this key. It's always implicitly loaded!
      (kk->flags & RLOOKUP_F_VAL_AVAILABLE) ||
      // There is no value in the sorting vector, and we don't need to load it from the document.
      ((kk->flags & RLOOKUP_F_SVSRC) && (RLookup_GetItem(kk, dst) == NULL))
  ));
```